### ISSUE TITLE
NFS shares not mounted, Vagrant trying to mount over wrong IP addresses

### ISSUE THREAD

**Issue by beaverusiv** [hashicorp/vagrant#9371]
### Vagrant version
Vagrant 2.0.1

### Host operating system
Kubuntu 17.04 Zesty

### Guest operating system
Ubuntu 16.04 Xenial

### Vagrantfile
```ruby
# -*- mode: ruby -*-
# vi: set ft=ruby :

VAGRANTFILE_API_VERSION = "2"
BASE_CODE_DIR = "/home/nicholas/code/"
WEB_PORT = 8080
NETWORK_INTERFACE = "enp0s31f6"

Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|

  config.vm.box = "ubuntu/xenial64"

  config.vm.hostname = "local-dev"
  config.vm.network :forwarded_port, id: "ssh", guest: 22, host: 2225
  config.vm.network :forwarded_port, id: "nginx", guest: 8082, host: WEB_PORT

  if defined? NETWORK_INTERFACE
    config.vm.network "public_network", :bridge => NETWORK_INTERFACE
  else
    config.vm.network "public_network"
  end

  config.vm.network "private_network", type: 'dhcp'

  config.vm.synced_folder "saltstack/", "/srv/saltstack", :create => true, :mount_options => ['dmode=777', 'fmode=777']
  config.vm.synced_folder BASE_CODE_DIR, "/mnt/code", :nfs => true, :nfs_udp => false, :linux__nfs_options => ['rw','sync','no_subtree_check','no_root_squash','nolock'], :mount_options => ['sec=sys'], :create => true
  config.vm.synced_folder "~", "/mnt/user", :create => true, :mount_options => ['dmode=777', 'fmode=777']

  config.vm.provider :virtualbox do |vb|
     vb.name = "SaltStack Development VM"
     vb.customize ["modifyvm", :id, "--memory", "2048"]
     vb.customize ["modifyvm", :id, "--natdnshostresolver1", "on"]
  end

  config.vm.provision "shell", path: "base-provision.sh"
end
```
base-provision.sh:
https://gist.github.com/beaverusiv/793b4ccf8d489619417aa7be8a0bb0f0

### Debug output
debug.log:
https://gist.github.com/beaverusiv/817787c8247cce6540b3fca920e15811
result of `ip addr`:
https://gist.github.com/beaverusiv/6983bc923c2ae28e602662443e158745
extra log lines added to /opt/vagrant/embedded/gems/gems/vagrant-2.0.1/plugins/providers/virtualbox/action/prepare_nfs_settings.rb:
https://gist.github.com/beaverusiv/b7ae1758203658d22d081dae734ec3e2

### Expected behavior
Vagrant should have mounted the nfs shares correctly and consistently

### Actual behavior
Vagrant complains about needing a host-only network. I believe it is picking up the wrong network when trying to mount the NFS shares and so discards the `machine_ip` after the check if it is in the range of the `host_ip`. According to `ip addr` it looks to be using the bridge network as the `machine_ip` when it should be getting the private DHCP network.

Vagrant does every now and then get it correct and successfully mount the shares but I have not been able to determine what affects it to make it succeed. I am not a Ruby developer but I can apply patches or try to get you any more info you may need.

**Comment by beaverusiv** [comment]
I destroyed the box and tried with a `ubuntu/trusty64` box which didn't work for unknown reasons (portmapping stuff). So I destroyed that one and remade a `ubuntu/xenial64` one. This time the output shows it gets the correct `machine_ip`:
https://gist.github.com/beaverusiv/8493843bd8d301550bf2838a76851041

It is now consistently finding the right one when I reload it. I have previously destroyed this box multiple times, I am wondering if when creating it VirtualBox sets the uuids which then can determine which order the adapters are listed and can determine the stability of finding the right IPs for NFS mounting.

**Comment by beaverusiv** [comment]
Ok, after a host reboot it is back to picking up the wrong adapter for the `machine_ip` (the bridge adapter: `br-281e7ab8fd71`).

I definitely think this is to do with the code that gets the IPs picking the wrong one.(https://github.com/hashicorp/vagrant/blob/master/plugins/providers/virtualbox/action/prepare_nfs_settings.rb#L39)

**Comment by jbonhag** [comment]
Hi there,

It looks like this has been resolved in the upcoming 2.2.8 release of Vagrant so I am now closing this issue. If you find that Vagrant 2.2.8 does not resolve the issue, please reopen this issue or create a new one.

Cheers!

**Comment by rgl** [comment]
@jbonhag, next time you close an issue can you add the commits that addressed the issue?

for this particular one, I'm not finding the commits... is it https://github.com/hashicorp/vagrant/commit/f593bb54d2fad54f2d79607679bc08b8859dc91f?
